src/app/clusters/mode-select-server/mode-select-delegate.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+*
+*    Copyright (c) 2023 Project CHIP Authors
+*    All rights reserved.
+*
+*    Licensed under the Apache License, Version 2.0 (the "License");
+*    you may not use this file except in compliance with the License.
+*    You may obtain a copy of the License at
+*
+*        http://www.apache.org/licenses/LICENSE-2.0
+*
+*    Unless required by applicable law or agreed to in writing, software
+*    distributed under the License is distributed on an "AS IS" BASIS,
+*    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*    See the License for the specific language governing permissions and
+*    limitations under the License.
+*/
+
+#include <app/clusters/mode-select-server/mode-select-server.h>
+#include "mode-select-delegate.h"
+
+using namespace std;
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ModeSelect;
+
+using ModeOptionStructType = Structs::ModeOptionStruct::Type;
+
+Status Delegate::HandleChangeToMode(uint8_t mode)
+{
+    return Status::Success;
+}
+
+void Delegate::HandleChangeToModeWitheStatus(uint8_t mode, ModeSelect::Commands::ChangeToModeResponse::Type &response)
+{
+    response.status = static_cast<uint8_t>(ChangeToModeResponseStatus::kSuccess);
```

Comment:
```suggestion
    response.status = to_underlying(ChangeToModeResponseStatus::kSuccess);
```
which will have the salutary effect of failing of the types don't match and it being impossible to cast to the wrong thing.